draft-vandijk-dprive-ds-dot-signal-and-pin/yardsticks/draft-ietf-dprive-phase2-requirements-01.md

```diff
@@ -0,0 +1,117 @@
+
+The draft is limited to DoT.
+We propose that other protocols update this draft by adding additional DNSKEY algorithms (TBD2 etc.).
+TODO: should we have words on future protocols such as DoQ?
+
+>         This SHALL include a
+>         method to publish a list of secure transport protocols (e.g.
+>         DoH, DoT and other future protocols not yet developed).  In
+>         addition this SHALL include whether a secure transport protocol
+>         MUST always be used (non-downgradable) or whether a secure
+>         transport protocol MAY be used on an opportunistic (not strict)
+>         basis.
+
+This draft specifies that the protocol indicated (DoT) is mandatory and no downgrades shall occur.
+
+>    8.   The authoritative domain owner or their administrator MUST have
+>         the option to vary their preferences on an authoritative
+>         nameserver to nameserver basis, due to the fact that
+>         administration of a particular DNS zone may be delegated to
+>         multiple parties (such as several CDNs), each of which may have
+>         different technical capabilities.
+
+This draft specifies that if not all authoritatives for a domain cooperate in DoT with the pinned keys, they must fail swiftly and not cause timeouts.
+
+>    9.   The specification of secure transport preferences MUST be
+>         performed using the DNS and MUST NOT depend on non-DNS
+>         protocols.
+
+Yes.
+
+>    10.  For the secure transport, TLS 1.3 (or later versions) MUST be
+>         supported and downgrades from TLS 1.3 to prior versions MUST not
+>         occur.
+
+Yes, but out of scope for this discovery/pinning draft.
+
+> 5.2.  Optional Requirements
+> 
+>    1.  QNAME minimisation SHOULD be implemented in all steps of
+>        recursion
+
+Out of scope.
+
+>    2.  DNSSEC validation SHOULD be performed
+
+Out of scope.
+
+>    3.  If an authoritative domain owner or their administrator indicates
+>        that (1) multiple secure transport protocols are available or
+>        that (2) a secure transport and insecure transport are available,
+>        then per the recommendations in [RFC8305] (aka Happy Eyeballs) a
+>        recursive server SHOULD initiate concurrent connections to
+>        available protocols.  Consistent with Section 2 of [RFC8305] this
+>        would be: (1) Initiation of asynchronous DNS queries to determine
+>        what transport protocols are supported, (2) Sorting of resolved
+>        destination transport protocols, (3) Initiation of asynchronous
+>        connection attempts, and (4) Establishment of one connection,
+>        which cancels all other attempts.
+
+We strongly disagree, with reference to [the bottom end of Duane Wessels' response to phase2-01](https://mailarchive.ietf.org/arch/msg/dns-privacy/eMtjqBBu6m7YhKEHkW5ZKfvZohg/).
+We also believe that when the presence of any secure transport is indicated, no insecure transport should ever be permitted.
```

Comment:
unrelated to the item this is answering to, but some thought on strict mode.

We currently have no way of signaling whether the zone owner wants to enable DoT in [strict or opportunistic mode](https://tools.ietf.org/html/rfc8310) when the recursive server supports this draft. The draft itself is aiming at strict though.

In the end, it is going to be on the recursive side to make a choice whether they want to be opportunistic or strict and there is nothing this draft can do about, e.g auth side is likely not going to track resovlers and deny cleartext service if the resolver was technically able to do DoT.
Asumming there is more than a couple of auth out there that start implementing this, It also seems logical from an operational point of view that one would be rolling out in opportunistic mode first, measure the number of times fallbacks need to happen, build operational knowledge and confidence before going into strict mode. So I would expect resolvers to provide an opportunistic mode (and metrics when falling back to cleartext) in order for people to give this a shot.

Reply:
This makes sense, and I think the model is identical to that for DNSSEC deployment. When a zone owner/operator decides to enable the security (DNSSEC or pinned DoT), they make a promise to do it right (have correct signatures/have a working port 853 with the right key). But resolver/validator operators indeed are in a position to 'see what happens' (permissive/strict validation / yes/no fallback from DoT [pinning]).

Textwise, I don't want to add nuance to every MUST in the document - so perhaps we can write one clear paragraph in the right place that explains the incremental deployment model for resolvers, perhaps as part of resolving your next comment.

Reply:
pushed, please check